src/Conversion/ONNXToTOSA/Tensor/Constant.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Constant.cpp - Const Op -----------------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX const operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/BuiltinAttributeInterfaces.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXConstOpLoweringToTOSA : public OpConversionPattern<ONNXConstantOp> {
+public:
+  using OpConversionPattern<ONNXConstantOp>::OpConversionPattern;
+  using OpAdaptor = typename ONNXConstantOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXConstantOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto valueAttr = adaptor.value();
+    auto sparseAttr = adaptor.sparse_value();
```

Comment:
Please use a concrete type instead of `auto`. I think it is `Optional<Attribute>`